fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/synthetic/SyntheticRecordPlannerTest.java

```diff
@@ -1040,6 +1041,225 @@ public void aggregateJoinIndex() throws Exception {
+        }
+
+
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore recordStore = recordStoreBuilder.setContext(context).open();
+
+            //force-mark the index as readable
+            recordStore.uncheckedMarkIndexReadable(indexName).get();
+
+            //now verify that no records exist in that index
+            Index joinIndex = recordStore.getRecordMetaData().getIndex("joinNestedConcat");
+            try (RecordCursor<IndexEntry> cursor = recordStore.scanIndex(joinIndex, IndexScanType.BY_VALUE, TupleRange.ALL, null, ScanProperties.FORWARD_SCAN)) {
+                Assertions.assertEquals(0, cursor.getCount().get(), "Wrote records to a disabled index!");
+            }
+        }
+    }
+
+    @Test
+    void updateSyntheticIndexesWhenInWriteOnly() throws Exception {
+        String indexName = addJoinedIndexToMetaData();
+
+        //write some data
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore recordStore = recordStoreBuilder.setContext(context).create();
+
+            TestRecordsJoinIndexProto.CustomerWithHeader customer = TestRecordsJoinIndexProto.CustomerWithHeader.newBuilder()
+                    .setHeader(TestRecordsJoinIndexProto.Header.newBuilder().setZKey(1).setIntRecId(1L))
+                    .setName("Scott")
+                    .setCity("Toronto")
+                    .build();
+            recordStore.saveRecord(customer);
+
+            TestRecordsJoinIndexProto.OrderWithHeader order = TestRecordsJoinIndexProto.OrderWithHeader.newBuilder()
+                    .setHeader(TestRecordsJoinIndexProto.Header.newBuilder().setZKey(1).setRecId("23"))
+                    .setOrderNo(10)
+                    .setQuantity(23)
+                    .setCustRef(TestRecordsJoinIndexProto.Ref.newBuilder().setStringValue("i:1"))
+                    .build();
+            recordStore.saveRecord(order);
+
+            context.commit();
+        }
+
+        //mark the index write only, then write some more data
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore recordStore = recordStoreBuilder.setContext(context).open();
+            recordStore.markIndexWriteOnly(indexName).get();
+
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore recordStore = recordStoreBuilder.setContext(context).open();
+            TestRecordsJoinIndexProto.CustomerWithHeader customer = TestRecordsJoinIndexProto.CustomerWithHeader.newBuilder()
+                    .setHeader(TestRecordsJoinIndexProto.Header.newBuilder().setZKey(1).setIntRecId(2L))
+                    .setName("Scott")
+                    .setCity("Toronto")
+                    .build();
+            recordStore.saveRecord(customer);
+
+            TestRecordsJoinIndexProto.OrderWithHeader order = TestRecordsJoinIndexProto.OrderWithHeader.newBuilder()
```

Comment:
I like this test, but I don't think it covers the case of updating a record while in write-only mode. It creates a second record. In order to cover updates, you'd have to save a record with the same header, but change the name or orderNo. And in order to actually cover updates on a write-only index you have to assert about the values of the index.
For this PR, I think the tests as you have them are probably fine, but we should add more coverage of indexes on synthetic types in various situations.
I was a bit thrown off because you have `updateSyntheticIndexesWhenInWriteOnly` and `deleteFromSyntheticIndexWhenIndexIsWriteOnly` so I thought the former was for updating a record, rather than simply updating the index.